code-challenges.test.js

```diff
@@ -64,4 +138,44 @@ const accountTransactions2 = [250, -89, 100, -96]
 const accountTransactions3 = []
 // Expected output: []
 
+describe("sumArray", () => {
+  it("returns an array of accumulating sum or an empty array", () => {
+    expect(sumArray(accountTransactions1)).toEqual([100, 83, 60, 51])
+    expect(sumArray(accountTransactions2)).toEqual([250, 161, 261, 165])
+    expect(sumArray(accountTransactions3)).toEqual([])
+  })
+})
+//Test failed as expected:  ReferenceError: sumArray is not defined
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 2 passed, 3 total
+
 // b) Create the function that makes the test pass.
+
+//Psuedo:
+//Create a function called sumArray, passing in an array as the input. 
+//Take the input array and use the .reduce method to calculate the accumulated values. 
+  //The reduce method requires certain parameters: 
+      //The reducer function exeutes on each element in an array and requires 2 parameters and can have optional ones.
+        //The 2 required parameters are the total and current value. 
+        //The 2 optional parameters are the current index and the array the values belong to. Initial value is also optional for a baseline value for the function to use. 
+  //In the reducer function, we will pass in sum, currentNum, and index as parameters. 
+    //sum will be the previous value returned
+    //By adding currentNum(current value) plus the number at index of sum at index - 1 (the previous accumulated value) OR 0
+    //Also, passing an empty array as an optional initial value. In this case, we also want the function to work on empty arrays. Since the reduce method does not work on empty arrays, having the optional parameter passed in, helped for this case. 
+//Output: An array of accumulating sum or an empty array. 
+
+const sumArray = (array) => {
+  // return array.map((value) => {     //Initial logic process was to map over the array and then add each element value to the previous sum.
+  //   return sum = 0, sum += value     //While this logic worked in theory, it ended up giving me incorrect values. 
+  // })
+  return array.reduce((sum, currentNum, index) =>     //Setup reducer function with sum, currentNum and index parameters
+    [...sum, currentNum + (sum[index - 1] || 0)], []) //Adding currentNum to sum of each previous value at each index
+}
```

Comment:
There is a couple different ways we can use reduce method. In our scenario, we essentially want it to iterate on each element in the array and perform math. We also want to keep track of each sum. The setup would be a little different if we were just interested in the "balance" at the end.